starknet/test/l1-execution.test.ts

```diff
@@ -0,0 +1,232 @@
+import dotenv from 'dotenv';
+import { starknet, ethers } from 'hardhat';
+import { Provider, Account, CallData, Uint256, uint256 } from 'starknet';
+import { safeWithL1AvatarExecutionStrategySetup } from './utils';
+
+dotenv.config();
+
+const eth_network = process.env.ETH_NETWORK_URL || '';
+const stark_network = process.env.STARKNET_NETWORK_URL || '';
+const account_address = process.env.ADDRESS || '';
+const account_pk = process.env.PK || '';
+
+function sleep(ms: number) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+describe('L1 Avatar Execution', function () {
+  this.timeout(1000000);
+
+  let signer: ethers.Wallet;
+  let safe: ethers.Contract;
+  let mockStarknetMessaging: ethers.Contract;
+  let l1AvatarExecutionStrategy: ethers.Contract;
+
+  // Using both a starknet hardhat and sn.js account wrapper as hardhat has cleaner deployment flow
+  // but sn.js has cleaner contract interactions. Syntax is being integrated soon.
+  let account: Account;
+  let accountSH: starknet.starknetAccount;
+  let starkTxAuthenticator: starknet.StarknetContract;
+  let vanillaVotingStrategy: starknet.StarknetContract;
+  let vanillaProposalValidationStrategy: starknet.StarknetContract;
+  let space: starknet.StarknetContract;
+  let ethRelayer: starknet.StarknetContract;
+
+  before(async function () {
+    accountSH = await starknet.OpenZeppelinAccount.getAccountFromAddress(
+      account_address,
+      account_pk,
+    );
+
+    account = new Account(
+      new Provider({ sequencer: { baseUrl: stark_network } }),
+      account_address,
+      account_pk,
+    );
+
+    const signers = await ethers.getSigners();
+    signer = signers[0];
+
+    const starkTxAuthenticatorFactory = await starknet.getContractFactory(
+      'sx_StarkTxAuthenticator',
+    );
+    const vanillaVotingStrategyFactory = await starknet.getContractFactory(
+      'sx_VanillaVotingStrategy',
+    );
+    const vanillaProposalValidationStrategyFactory = await starknet.getContractFactory(
+      'sx_VanillaProposalValidationStrategy',
+    );
+    const ethRelayerFactory = await starknet.getContractFactory('sx_EthRelayerExecutionStrategy');
+    const spaceFactory = await starknet.getContractFactory('sx_Space');
+
+    try {
+      // If the contracts are already declared, this will be skipped
+      await accountSH.declare(starkTxAuthenticatorFactory);
+      await accountSH.declare(vanillaVotingStrategyFactory);
+      await accountSH.declare(vanillaProposalValidationStrategyFactory);
+      await accountSH.declare(ethRelayerFactory);
+      await accountSH.declare(spaceFactory);
+    } catch {}
```

Comment:
There is `declareIfNot` available for this.

Reply:
oh thanks will update